src/plugins/workspace/public/components/workspace_picker_content/workspace_picker_content.tsx

```diff
@@ -38,20 +52,49 @@ export const WorkspacePickerContent = ({
   onClickWorkspace,
 }: Props) => {
   const workspaceList = useObservable(coreStart.workspaces.workspaceList$, []);
+  const isDashboardAdmin = coreStart.application.capabilities?.dashboards?.isDashboardAdmin;
   const availableUseCases = useObservable(registeredUseCases$, []);
+  const [search, setSearch] = useState('');
 
-  const filteredWorkspaceList = useMemo(() => {
-    return workspaceList.slice(0, MAX_WORKSPACE_PICKER_NUM);
-  }, [workspaceList]);
+  const debouncedSetQueryInput = useMemo(() => {
+    return debounce(setSearch, 100);
```

Comment:
We should debounce the async search instead of the `setSearch`, if we debounce the setSearch API, I guess user will experience a 100ms delay for every input.

Reply:
And actually we are doing the search without calling any API, I do not think a debounce here is necessary.